backend/src/main/java/com/happy/friendogly/footprint/service/FootprintCommandService.java

```diff
@@ -89,4 +93,14 @@ public UpdateWalkStatusResponse updateWalkStatus(Long memberId, UpdateWalkStatus
         footprint.updateWalkStatusWithCurrentLocation(new Location(request.latitude(), request.longitude()));
         return new UpdateWalkStatusResponse(footprint.getWalkStatus());
     }
+
+    public void stopWalking(Long memberId, StopWalkingRequest request) {
+        Member member = memberRepository.getById(memberId);
+        Footprint footprint = footprintRepository.getById(request.footprintId());
+        if (footprint.getMember() != member) {
+            throw new FriendoglyException("본인의 발자국만 변경 가능합니다.", NOT_ALLOW_OTHER_FOOTPRINT_CHANGE, BAD_REQUEST);
+        }
```

Comment:
```suggestion
        Footprint footprint = footprintRepository.getById(request.footprintId());
        if (!footprint.isCreatedBy(memberId)) {
            throw new FriendoglyException("본인의 발자국만 변경 가능합니다.", NOT_ALLOW_OTHER_FOOTPRINT_CHANGE, BAD_REQUEST);
        }
```

Footprint 안에 있는 `isCreatedBy()` 메서드 쓰면 더 객체지향적으로 작성 가능하겠네요